<?xml version="1.0" encoding="utf-8"?>

<launch>
    <!-- arguments -->
    <!-- <arg name="nodelet_manager" default="tloam_nodelet_manager" /> -->
    <arg name="cloud_topic" default="/velodyne_points" />

    <!-- <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager)" args="manager" output="screen"/> -->

    <!-- kitti_reader_nodelet -->
    <node pkg="tloam" type="kitti_reader_node" name="kitti_reader_node" output="screen">
        <param name="start_frame" value="0" />
    </node>

    <!-- segmentation_nodelet -->
    <node pkg="tloam" type="segmentation_node" name="segmentation_node" output="screen"></node>

    <!-- lidar_odometry_nodelet -->
    <!-- launch-prefix="gdb -ex run - -args"  -->
    <node pkg="tloam" type="lidar_odometry_node" name="lidar_odometry_node" output="screen"></node>

    <!-- Load rviz_car_model before rviz  x y z qx qy qz qw-->
    <include file="$(find tloam)/launch/robot_car.launch">
        <arg name="fixed_frame" value="base_link" />
        <arg name="transform" value="0.0 0.0 0.0  0.0 0.0 0.0 1.0" />
    </include>

    <!-- rviz -->
    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find tloam)/rviz/tloam.rviz" />
    </group>

</launch>